### PR TITLE
Support for minigraph generation for packet based chassis.

### PR DESCRIPTION
**Pull request by abdosi**
What I did:

Below changes goes with the sonic-buildimage PR: https://github.com/Azure/sonic-buildimage/pull/8966

    1. Changes for Generation LC-Graph for packet-based chassis.
    2. Added Support Ipv6 Peering on Loopback4096 for voq also
    3. Updated asic topology yml files to be offset of slot
    4. Made slot_num to take string slot<number> instead of number
    5. Consolidated template_dpg_voq_asic.j2 into dpg_asic.j2
    6. Remove Loopback4096 from asic topology and parse as dut invertory for
       multi-asic
    7. Updated topo_facts parsing for asic topology_
    8. Internal BGP Session rename from <VoqChassisInternal> to <ChassisInternal> and take switch_type as value.
   
 

How I Verify
1. Verified gen-mg for multi-asic/voq-chassis/packet-chassis dut's.

### CODE REVIEW

File: ansible/library/port_alias.py [hunk 1]
```diff
@@ -78,20 +78,22 @@ def get_platform_type(self):
...
         if platform is None:
             return None
-        if asic_id is None:
+        if asic_id is None or asic_id == '':
             portconfig = os.path.join(FILE_PATH, platform, self.hwsku, PORTMAP_FILE)
-        else:
+        elif slotid is None or slotid == '':
             portconfig = os.path.join(FILE_PATH, platform, self.hwsku, str(asic_id), PORTMAP_FILE)
+        else:
+            portconfig = os.path.join(FILE_PATH, platform, self.hwsku, str(slotid), str(asic_id), PORTMAP_FILE)
```
Comment (sanmalho-git) [inline ansible/library/port_alias.py:90]: Is the port_config.ini file different for a card with the same hwsku being in a different slot?

File: ansible/lab [hunk 2]
```diff
@@ -148,3 +150,39 @@ sonic_multi_asic_2:
     vlab-08:
       ansible_host: 10.250.0.112
       ansible_hostv6: fec0::ffff:afa:c
+
+sonic_msft_sup:
+  vars:
+    HwSku:  msft-RP-O
+    slot_num: slot0
```
[the remark this answers is not in the document]
Comment (abdosi) [reply ansible/lab:157]: Have fixed in latest commit to strip "slot" from the string.

**Comment by abdosi** [comment]
@arlakshm can you review/approve this.

**Comment by sanmalho-git** [comment]
@abdosi - sorry, but am just getting to test this PR out on Nokia's VoQ chassis. 

I see two issues in the minigraph generated with this PR:

- In Cpg
   - Missing BGPRouterDeclaration section for remote asics - <remote_linecard>-ASIC<#>
   - For multi-asic linecard, missing BGPRouterDeclaraion section for my asics - ASIC<#>
- In Dpg, we are missing the DeviceDataPlaneInfo for my asics in a multi-asic linecard
   - This is generated in template_dpg_asic, which is dependent on asic_topo_config. For voq_chassis, asic_topo_config is empty dictionary - as the iBGP connections depend on the what linecards are present in the chassis, and also what asics are on each linecard. This was the reason that I had created a seperate minigraph_dpg_voq template.

**Comment by abdosi** [comment]
> @abdosi - sorry, but am just getting to test this PR out on Nokia's VoQ chassis.
> 
> I see two issues in the minigraph generated with this PR:
> 
> * In Cpg
>   
>   * Missing BGPRouterDeclaration section for remote asics - <remote_linecard>-ASIC<#>
>   * For multi-asic linecard, missing BGPRouterDeclaraion section for my asics - ASIC<#>
> * In Dpg, we are missing the DeviceDataPlaneInfo for my asics in a multi-asic linecard
>   
>   * This is generated in template_dpg_asic, which is dependent on asic_topo_config. For voq_chassis, asic_topo_config is empty dictionary - as the iBGP connections depend on the what linecards are present in the chassis, and also what asics are on each linecard. This was the reason that I had created a seperate minigraph_dpg_voq template.

@sanmalho-git thanks for validating. we will check and fix this by this week.

cc @SuvarnaMeenakshi

**Comment by sanmalho-git** [comment]
My approach to fixing this is to

-	create an almost empty asic topology file for our multi-asic linecard with slot0.
-	In config_sonic_basedon_testbed.yml,  when dealing with ‘voq’ switch_type, before running through the Jinja2 templates, change ‘slot0’ in the asic_topo_config to the ‘slot_num’ that is defined in the inventory.

I have prototyped with this solution, and works fine for both single and multi-asic linecards in a VoQ chassis. 

Below is the asic topology for our multi-asic linecard Nokia_IXR7250e_36x400G card (in vars/topo_Nokia_IXR7250e_36x400G.yml) that I prototyped with:

```
slot0:
    ASIC0:
      topology:
        NEIGH_ASIC:
      configuration_properties:
        common:
          asic_type: FrontEnd 
      configuration:

    ASIC1:
      topology:
        NEIGH_ASIC:
      configuration_properties:
        common:
          asic_type: FrontEnd
      configuration:
```

Please let me know if you are OK with the approach

**Comment by abdosi** [comment]
> My approach to fixing this is to
> 
> * create an almost empty asic topology file for our multi-asic linecard with slot0.
> * In config_sonic_basedon_testbed.yml,  when dealing with ‘voq’ switch_type, before running through the Jinja2 templates, change ‘slot0’ in the asic_topo_config to the ‘slot_num’ that is defined in the inventory.
> 
> I have prototyped with this solution, and works fine for both single and multi-asic linecards in a VoQ chassis.
> 
> Below is the asic topology for our multi-asic linecard Nokia_IXR7250e_36x400G card (in vars/topo_Nokia_IXR7250e_36x400G.yml) that I prototyped with:
> 
> ```
> slot0:
>     ASIC0:
>       topology:
>         NEIGH_ASIC:
>       configuration_properties:
>         common:
>           asic_type: FrontEnd 
>       configuration:
> 
>     ASIC1:
>       topology:
>         NEIGH_ASIC:
>       configuration_properties:
>         common:
>           asic_type: FrontEnd
>       configuration:
> ```
> 
> Please let me know if you are OK with the approach

Thanks sanmalho-git. this looks fine to me . I assume this is for Problem#2. What about Problem#1

**Comment by sanmalho-git** [comment]
For problem#1, there are some minor fixes in minigraph_cpg.j2.

Also, if I can get the command to generate minigraph of the packet-based chassis topology, that would help me test the solution out for packet based chassis as well.

**Comment by abdosi** [comment]
> For problem#1, there are some minor fixes in minigraph_cpg.j2.
> 
> Also, if I can get the command to generate minigraph of the packet-based chassis topology, that would help me test the solution out for packet based chassis as well.

thanks @sanmalho-git Can you create PR for now ?  I will take your change and verify. Not sure current public code will work to generate chassis-packet template. Will create seprate to make it work if not working.

**Comment by sanmalho-git** [comment]
@abdosi @SuvarnaMeenakshi - PR https://github.com/Azure/sonic-mgmt/pull/4699 has been raised to address the issues with minigraph generation for linecards in a VoQ chasiss. Please review and also add the 'chassis' label to the PR.

**Pull request merged**